solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -1086,6 +1099,11 @@ private void createEphemeralLiveNode() throws KeeperException,
     log.info("Register node as live in ZooKeeper:{}", nodePath);
     List<Op> ops = new ArrayList<>(2);
     ops.add(Op.create(nodePath, null, zkClient.getZkACLProvider().getACLsToAdd(nodePath), CreateMode.EPHEMERAL));
+
+    // Create the roles node as well
+   cc.nodeRoles.getRoles().forEach((role, val) -> ops.add(Op.create(ZkStateReader.NODE_ROLES + "/" + role.roleName+ "/"+val +"/"+ nodeName,
```

Comment:
Another example of ZK path assembly that would benefit from a method building that path (see comment line 862)

Reply:
Fixed, thanks.